app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -43,9 +43,9 @@ def images(images)
       uid      = image['image_id']
       location = image['image_location']
       name     = get_from_tags(image, :name)
-      name     ||= image['name']
-      name     ||= $1 if location =~ /^(.+?)(\.(image|img))?\.manifest\.xml$/
-      name     ||= uid
+      name     = image['name'] if name.blank?
+      name     = $1 if name.blank? && location =~ /^(.+?)(\.(image|img))?\.manifest\.xml$/
+      name     = uid if name.blank?
 
```

Comment:
@djberg96 - what if `get_from_tags` returns `nil`?

Reply:
Then it resorts to the actual name property. I'm not sure why the Name tag gets precedent, but that behavior was here when I got here.